aten/src/ATen/native/cuda/Gesv.cu

```diff
@@ -138,6 +160,35 @@ std::tuple<Tensor,Tensor> _gesv_helper_cuda(const Tensor& self, const Tensor& A)
   return std::tuple<Tensor,Tensor>(b_working_copy, A_working_copy);
 }
 
+std::tuple<Tensor&,Tensor&> _gesv_single_out_cuda(Tensor& result0, Tensor& result1,
+    const Tensor& self, const Tensor& A) {
+#ifndef USE_MAGMA
+AT_ERROR("gesv: MAGMA library not found in "
+    "compilation. Please rebuild with MAGMA.");
+#else
+  int64_t bx = self.size(0);
+  int64_t by = (self.dim() == 1) ? 1 : self.size(1);
+  int info;
+  int* ipiv;
+
+  result1 = A.t().clone();
```

Comment:
What is the `.t().clone()` for? Is this to ensure result1, result0 are in column major format?

If so, this code assumes that A, self are contiguous and they might not be

Reply:
edit: I thought about it some more and `.t().clone()` is fine.

You could call https://github.com/pytorch/pytorch/blob/71626491c40c86eb1ee410e695c467486084144c/aten/src/ATen/native/LinearAlgebraUtils.h#L15 instead (and probably rename the function to something like "cloneColumnMajor") so this becomes clearer.